PyDynamic/uncertainty/interpolate.py

```diff
@@ -303,6 +303,84 @@ def interp1d_unc(
+        extrap_range_above = t_new > np.max(t)
+        extrap_range = extrap_range_below | extrap_range_above
+        # .. and inside t's bounds.
+        interp_range = ~extrap_range
+
+        # Initialize the result array for the standard uncertainties.
+        uy_new = np.empty_like(y_new)
+
+        # Initialize the sensitivity matrix of shape (M, N)
+        C = np.zeros((len(t_new), len(uy)), "float64")
+
+        # First extrapolate the according values if required and then
+        # compute interpolated uncertainties following White, 2017.
+
+        # If extrapolation is needed, fill in the values provided via fill_unc.
+        if np.any(extrap_range):
+            # At this point fill_unc is either a float (np.nan is a float as well) or
+            # a 2-tuple of floats. In case we have one float we set uy_new to this value
+            # inside the extrapolation range.
+            if isinstance(fill_unc, float):
+                uy_new[extrap_range] = fill_unc
+            else:
+                # Now fill_unc should be a 2-tuple, which we can fill into uy_new.
+                uy_new[extrap_range_below], uy_new[extrap_range_above] = fill_unc
+
+            if returnC:
+                # In each row of C corresponding to an extrapolation value below the
+                # original range set the first column to 1 and in each row of C
+                # corresponding to an extrapolation value above the original range set
+                # the last column to 1. It is important to do this before
+                # interpolating, because in general those two columns can contain
+                # non-zero values in the interpolation range.
+                C[:, 0], C[:, -1] = extrap_range_below, extrap_range_above
+
+        # If interpolation is needed, compute uncertainties following White, 2017.
+        if np.any(interp_range):
+            # This following section is taken mainly from scipy.interpolate.interp1d to
+            # determine the indices of the relevant original timestamps (or frequencies)
+            # just for the interpolation range.
+            # --------------------------------------------------------------------------
+            # 2. Find where in the original data, the values to interpolate
+            #    would be inserted.
+            #    Note: If t_new[n] == t[m], then m is returned by searchsorted.
+            t_new_indices = np.searchsorted(t, t_new[interp_range])
+
+            # 3. Clip x_new_indices so that they are within the range of
+            #    self.x indices and at least 1.  Removes mis-interpolation
+            #    of x_new[n] = x[0]
+            t_new_indices = t_new_indices.clip(1, len(t) - 1).astype(int)
+
+            # 4. Calculate the uncertainty
+            # generate spline of sensitivities, this procedure is described by eq. (19) of White2017
+            F = []
+            for i in range(len(t)):
+                tf = t
+                xf = np.zeros_like(t)
```

Comment:
I am wondering, what does the 'f' in `xf` stand for?

Reply:
It probably was because it is the `x` which is necessary for the calculation of the `F`, hence `xf`. But I really like your suggestion and renaming it to `x_temp`.